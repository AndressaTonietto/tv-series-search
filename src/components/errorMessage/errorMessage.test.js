import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";

import ErrorMessage from "./errorMessage";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

const testErrorMessage = "test error message";
describe("ErrorMessage", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Router>
          <ErrorMessage error={testErrorMessage} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders the error message and the link to home page correctly", () => {
    render(
      <Router>
        <ErrorMessage error={testErrorMessage} />
      </Router>
    );

    expect(
      screen.getByText((content, element) => content.includes(testErrorMessage))
    ).toBeInTheDocument();
    expect(screen.getByTestId("home-page-link")).toHaveAttribute("href", "/");
  });
});
