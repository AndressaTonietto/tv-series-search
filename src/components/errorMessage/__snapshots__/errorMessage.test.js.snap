// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders correctly 1`] = `
<div
  className="toast toast-center min-w-full"
>
  <div
    className="alert alert-error text-wrap"
  >
    <span>
      test error message
      . Please try again or go to our
       
      <a
        className="link"
        data-testid="home-page-link"
        href="/"
        onClick={[Function]}
      >
        home page
      </a>
      .
    </span>
  </div>
</div>
`;
